MyString.java

```diff
@@ -49,8 +72,16 @@ public static boolean subsetOf(String str1, String str2) {
      * @return a string consisting of the characters of str, separated by spaces.
      */
     public static String spacedString(String str) {
-        //// Replace the following statement with your code
-        return null;
+        String spacedString = "";
+        for (int i=0;i< str.length();i++) {
+            if (i == str.length()-1) {
+                spacedString += str.charAt(i);
+            }
+            else {
+            spacedString += str.charAt(i) + " ";
+            }

```

Comment:
if you have the same code segment in both if and else it usually means that it can be moved outside the if
consider the following:
```
spacedString += str.charAt(i);
if (i == str.length()-1) {

} else {
                spacedString +=  " ";
}
```
consider the following:
but it is still not good since I dont have anything in if, but do have in else, so we can switch the condition such that else case will be in new if. 
```
spacedString += str.charAt(i);
if (i != str.length()-1) {
                spacedString +=  " ";
}
```